corehq/apps/locations/resources/v0_6.py

```diff
@@ -43,31 +43,27 @@ def dehydrate(self, bundle):
         return bundle
 
     def obj_create(self, bundle, **kwargs):
-        domain = bundle.data['domain']
+        if 'domain' not in bundle.data or 'name' not in bundle.data:
```

Comment:
You should be able to get domain from the kwargs since it's part of the URL mapping.